integration/util/waysnodes/nodes2wayblotdb/db.go

```diff
@@ -0,0 +1,214 @@
+// Package nodes2wayblotdb stores `fromNodeID,toNodeID -> wayID` mapping in blotdb.
+package nodes2wayblotdb
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/golang/glog"
+
+	bolt "go.etcd.io/bbolt"
+)
+
+// DB stores `fromNodeID,toNodeID -> wayID` in blotdb.
+type DB struct {
+	db *bolt.DB
+}
+
+const (
+	defaultBucket = "bucket"
+)
+
+var (
+	errEmptyDB     = errors.New("empty db")
+	errKeyNotFound = errors.New("key not found")
+)
+
+// Open creates/opens a DB structure to store the nodes2way mapping.
+func Open(dbFilePath string, readOnly bool) (*DB, error) {
+	var db DB
+
+	options := bolt.Options{
+
+		// Default Bolt Options
+		Timeout:      0,
+		NoGrowSync:   false,
+		FreelistType: bolt.FreelistArrayType,
+
+		// set readonly
+		ReadOnly: readOnly,
+	}
+
+	var err error
+	db.db, err = bolt.Open(dbFilePath, 0666, &options)
+	if err != nil {
+		return nil, err
+	}
+
+	if readOnly {
+		return &db, nil
+	}
+
+	// to improve write performance, but will manually sync before close
+	db.db.NoSync = true
```

Comment:
I assume `NoSync` could improve performance a lot in our case, do you have a draft value?

I think the benefit of `NoSync` is `single commit`, means no db.log in the middle and let OS to arrange write throughput.  By the way, I think bboltdb will open bucket file by `mmap`, even set to `NoSync`, but physical memory usage should not as large as total data we plan to write?

But it says that `NoSync` is dangerous, might corrupt db's metadata.  Is it only suitable for following cases:
- Our case here, single bucket, and after pre-processing, all data has been written into db, even if anything wrong, just rebuild.  If there is possibility that we make a common module of `db` with multiple `bucket`, put `NoSync` as parameter might be better.
- Bulk load all data from bucket

Reply:
The comments for `NoSync` explains clearly(https://pkg.go.dev/go.etcd.io/bbolt?tab=doc#DB:): 
```go
	// Setting the NoSync flag will cause the database to skip fsync()
	// calls after each commit. This can be useful when bulk loading data
	// into a database and you can restart the bulk load in the event of
	// a system failure or database corruption.
```
I think it's very smililar with the `fwrite()`: the default behavior is let OS decide when to sync to disk for performance, and users have to manually `fsync()` if want to make sure all contents are committed.(Maybe some implementation difference that I haven't figured out yet.)      
I haven't test the performance yet. Currently the build db process is very slow. I'll profile to improve the performance later.